CST338/Seperate files/assig3.java

```diff
@@ -108,7 +113,94 @@ public static void main(String[] args) {
       }
       System.out.println("End of dealing shuffled cards");
       
+      System.out.println("\n--------Deck & Hand Interaction--------");
+      System.out.println("------Unshuffled Deck------");
+      // Declare P4 variables
+      int numPlayers, numCards;
       
+      final int MIN_PLAYERS = 1;
+      final int MAX_PLAYERS = 10;
+
+      boolean validInput = false;
+      Scanner userInput = new Scanner(System.in);  // Instantiate scanner object
+      
+      do
+      {
+         // Request user input
+         System.out.print("Enter number of players in game (1-10): ");
+         numPlayers = userInput.nextInt();
+
+         // Verify input
+         if (numPlayers >= MIN_PLAYERS && numPlayers <= MAX_PLAYERS)
+            
+            // Break loop if user input is good
+            validInput = true;
+
+      } while (!validInput);
       
+      // Instantiate game objects
+      Deck deck = new Deck();                   // Single deck, not shuffled
+      Hand players[] = new Hand[numPlayers];    // Array containing player hands
+      
+      // Give each player a hand
+      for (int i = 0; i < numPlayers; i++)
+         players[i] = new Hand();
+      
+      // Get number of cards in deck and store it
+      numCards = deck.topCard();                
+
+      // Deal cards to players
+      for (int i = 0; i <= numCards; i++)
+      {
+         // Get a card from top of deck
+         Card tmpCard = deck.dealCard();
+
+         // Check if card is valid
+         if (!tmpCard.getErrorFlag())
+         {
+            // Give card to player
+            players[i % players.length].takeCard(tmpCard);
+         }
+      }
+
+      // Output players' hands
+      for (int i = 1; i <= players.length; i++)
+      {
+         System.out.println("Player " + i + " Hand: ");
+         System.out.println(players[i-1].toString());
+      }
+
+      // Get cards back from players
+      for (int i = 0; i < players.length; i++)
+         players[i].resetHand();
+      
+      // Reset and shuffle deck
+      deck.init(1);
```

Comment:
shouldn't this be 
deck.init(numPlayers)
or even better, 
deck = new Deck(numPlayers)

Reply:
The function prototype for init is init(numPacks).